src/app/components/harvest/screens/metadata-review/metadata-review.component.ts

```diff
@@ -248,27 +248,41 @@ export class MetadataReviewComponent
       : row.harvestItem?.path;
 
   public updateHarvestWithMappingChange(): void {
-    const mappings: HarvestMapping[] = [];
+    // create a new "temporary" model of the Harvest mappings
+    const newMappings: HarvestMapping[] = this.harvest.mappings;
 
     /*
      * Iterate through all the rows, extract folders which have mappings and
      * push that list to the mappings array. We could use a filter and map for
      * this, however it's likely less performant
      */
     this.rows.forEach((row): void => {
+
       if (row.rowType === RowType.folder && isInstantiated(row.mapping)) {
-        mappings.push(row.mapping);
+        // since the path attribute is a unique identifier on a mapping, we can safely assume
+        // that there will ever only be one mapping returned from the bellow filter
+        const newMapping = newMappings.filter(mapping => mapping.path === row.mapping.path)[0];
+        if (isInstantiated(newMapping)) {
+          Object.assign(newMapping, row.mapping);
+        } else {
+          // the mapping doesn't exist on the current harvest model, so create a new one
+          newMappings.push(row.mapping);
+        }
+
```

Comment:
i just realised this would be a *lot* simpler if we used a `Map<string, HarvestMapping>` as the temporary collection for `newMappings`